### PR TITLE
Reimplementing PR 88 with corrections for underlying code drift

### PR DESCRIPTION
**Pull request by bc-bfenton**
Also...
More doc cleanup
Removed unused "isComplete" property
Split auth methods into basic and OAuth versions

### CODE REVIEW

File: src/Bigcommerce/Api/Client.php [hunk 1]
```diff
@@ -57,6 +57,58 @@ class Client
...
+     * Accepts both OAuth and Basic Auth credentials
+     *
+     * @param array $settings
+     */
+    public static function configure($settings)
+    {
+        if (isset($settings['client_id'])) {
+            self::configureOAuth($settings);
+        } else {
+            self::configureBasicAuth($settings);
```
Comment (aleachjr) [inline src/Bigcommerce/Api/Client.php]: Per the requirements, this should be removed. PM intentionally only wants to support oAuth on this latest version as we will be removing legacy API support sometime this year.

Comment (bc-bfenton) [inline src/Bigcommerce/Api/Client.php]: I've heard conflicting things on this from multiple people. @mattolson thoughts? do we need to rope in PM?

Comment (aleachjr) [inline src/Bigcommerce/Api/Client.php]: I confirmed with PM that the intent is to remove support for basic auth from the client.

Comment (aleachjr) [inline src/Bigcommerce/Api/Client.php]: Just for clarity, we are going to leave in basic auth until we have oauth support for private apps.

**Comment by aleachjr** [comment]
:+1:

**Pull request merged**